echopop/nwfsc_feat/stratified.py

```diff
@@ -0,0 +1,1002 @@
+    with bootstrap resampling for uncertainty estimation. It handles virtual transect
+    creation, stratified bootstrap sampling, and confidence interval estimation.
+
+    Parameters
+    ----------
+    model_parameters : Dict[str, Any]
+        Dictionary containing model configuration parameters:
+        - "transects_per_latitude": Number of transects per degree latitude
+        - "strata_transect_proportion": Proportion of transects to sample per stratum
+        - "num_replicates": Number of bootstrap replicates
+    resample_seed : int, optional
+        Random seed for reproducible bootstrap resampling, by default None.
+
+    Attributes
+    ----------
+    model_params : Dict[str, Any]
+        Stored model parameters.
+    rng : np.random.Generator
+        Random number generator for bootstrap sampling.
+    resampled_df : pd.DataFrame or None
+        DataFrame containing bootstrap replicates (set after stratified_bootstrap).
+    variable : str or None
+        Name of the response variable being analyzed.
+    transect_summary : pd.DataFrame
+        Summary statistics for each transect within strata.
+    strata_summary : pd.DataFrame
+        Summary statistics for each stratum.
+    survey_summary : Dict[str, pd.DataFrame]
+        Dictionary containing survey-level summaries for strata and overall survey.
+
+    Notes
+    -----
+    The Jolly-Hampton method is commonly used in fisheries acoustic surveys for
+    estimating fish biomass and abundance with stratified sampling designs.
+
+    References
+    ----------
+    Jolly, G.M., and Hampton, I. (1990). A stratified random transect design for
+    acoustic surveys of fish stocks. *Canadian Journal of Fisheries and Aquatic Sciences*,
+    *47*(7), 1282-1291. https://doi.org/10.1139/f90-147
+
+    Examples
+    --------
+    >>> model_params = {
+    ...     "transects_per_latitude": 5,
+    ...     "strata_transect_proportion": 0.75,
+    ...     "num_replicates": 100,
+    ... }
+    >>> analysis = JollyHampton(model_params, resample_seed=42)
+    >>> virtual_data = analysis.create_virtual_transects(
+    ...     mesh_data, geostratum_df,
+    ...     stratify_by=["geostratum_inpfc"],
+    ...     variable="biomass"
+    ... )
+    >>> analysis.stratified_bootstrap(
+    ...     virtual_data,
+    ...     stratify_by=["geostratum_inpfc"],
+    ...     variable="biomass"
+    ... )
+    >>> summary = analysis.summarize(ci_percentile=0.95)
+    """
+
+    def __init__(
+        self,
+        model_parameters: Dict[str, Any],
+        resample_seed: Optional[int] = None,
+    ):
+        # Ingest model parameters
+        self.model_params = model_parameters
+
+        # Initialize the random number generator
+        self.rng = np.random.default_rng(resample_seed)
+
+        # Initialize attributes
+        self.resampled_df = None
+        self.variable = None
+        self.transect_summary = None
+        self.strata_summary = None
+        self.survey_summary = None
```

Comment:
Could you add in comment here what each attribute stores? `resampled_df` is the main one that triggers this request - I feel there's a more suitable name given the content (so similarly I feel there is probably another name for `_prepare_resampled_dataframe`).

Reply:
maybe something like:
- `self.resampled_df` -> `self.bootstrap_results`
- `self. _prepare_resampled_dataframe` -> `self._format_boostrap_results`